src/streaming/TextSourceBuffer.js

```diff
@@ -91,37 +138,33 @@ MediaPlayer.dependencies.TextSourceBuffer = function () {
                 bytes = new Uint8Array(bytes);
                 ccContent=window.UTF8.decode(bytes);
                 try {
-                    result = self.getParser().parse(ccContent);
-                    label = mediaInfo.id;
-                    lang = mediaInfo.lang;
-                    self.getTextTrackExtensions().addTextTrack(self.videoModel.getElement(), result, label, lang, true);
-                    self.eventBus.dispatchEvent({type:MediaPlayer.events.TEXT_TRACK_ADDED});
+                    result = self.getParser(mimeType).parse(ccContent);
+                    createTextTrackFromMediaInfo(result, mediaInfo);
                 } catch(e) {
                     self.errHandler.closedCaptionsError(e, "parse", ccContent);
                 }
             }
         },
 
+        getIsDefault:function(mediaInfo){
+            return mediaInfo.index === this.mediaInfos[0].index; //TODO How to tag default. currently same order as in manifest. Is there a way to mark a text adaptation set as the default one?
+        },
+
         abort:function() {
-            this.getTextTrackExtensions().deleteCues(this.videoModel.getElement());
+            this.videoModel.getElement().textTracks.removeEventListener('change', onTextTrackChange);
+            this.textTrackExtensions.deleteAllTextTracks();
```

Comment:
Good point.  I created this class at the beginning and Envivo added to it with the append call and fragmented text support.  Not sure if the intention was a true polyfill but seems it went that way over the course of time.   I'll move the stuff in abort to reset but if abort is to cancel the current fragment not sure what it will do in this class as there is nothing to cancel at this level.